contracts/BaseDocumentStore.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.6.10;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+
+contract BaseDocumentStore is Initializable {
+  string public name;
+  string public version;
+
+  /// A mapping of the document hash to the block number that was issued
+  mapping(bytes32 => uint256) public documentIssued;
+  /// A mapping of the hash of the claim being revoked to the revocation block number
+  mapping(bytes32 => uint256) public documentRevoked;
+
+  event DocumentIssued(bytes32 indexed document);
+  event DocumentRevoked(bytes32 indexed document);
+
+  function initialize(string memory _name) public initializer {
+    version = "2.3.0";
+    name = _name;
+  }
+
+  function issue(bytes32 document) public virtual onlyNotIssued(document) {
+    documentIssued[document] = block.number;
+    emit DocumentIssued(document);
+  }
+
+  function bulkIssue(bytes32[] memory documents) public {
+    for (uint256 i = 0; i < documents.length; i++) {
+      issue(documents[i]);
+    }
+  }
```

Comment:
Vulnerability?

Can someone not the owner issue a document now?

Reply:
Will suggest to:
1. mark all these function `internal` as oppose to `virtual` since the access controls are incomplete and needs to be called by implementing contract.
2. change the names to `_issue` `_bulkIssue` etc. See ERC20 pattern for minting. The base contract for mintable token has the function `_mint` and then implementing contract will then decide how it wants to use it (if at all).

Reply:
Yup I went to see the erc20 -> burnable/capped/etc example was good to show inheritance!